### PR TITLE
Fix calls to hasOwnProperty

### PR DESCRIPTION
**Pull request by taion**
Replaces #3201

### CODE REVIEW

File: modules/matchRoutes.js [hunk 1]
```diff
@@ -35,9 +35,7 @@ function getIndexRoute(route, location, callback) {
       callback(error, !error && createRoutes(indexRoute)[0])
     })
   } else if (route.childRoutes) {
-    const pathless = route.childRoutes.filter(function (obj) {
-      return !obj.hasOwnProperty('path')
-    })
+    const pathless = route.childRoutes.filter(childRoute => !childRoute.path)
```
Comment (taion) [inline modules/matchRoutes.js]: This is just a cleaner way to do that check. I considered writing this as

```js
({ path }) => !path
```

but thought using the explicit `childRoute` as the argument name was clearer.

File: modules/createTransitionManager.js [hunk 2]
```diff
@@ -104,7 +104,7 @@ export default function createTransitionManager(history, routes) {
     return route.__id__ || create && (route.__id__ = RouteGuid++)
   }
 
-  const RouteHooks = {}
+  const RouteHooks = Object.create(null)
```
Comment (taion) [inline modules/createTransitionManager.js]: This is an internal-only object used as a map, so it's more correct to use a `null`-prototype object than a full-fledged object as a map. There's no concern with breaking any consumers, as `RouteHooks` is not visible outside this module.

**Comment by taion** [comment]
@ryanflorence IMO this would justify releasing a v2.0.2, except I think we might need https://github.com/mjackson/history/pull/253 as well.

**Comment by naoufal** [comment]
@taion Do you know which release introduced this bug?

**Comment by taion** [comment]
Nothing on our side per se – the logic has always worked this way. The issue was exposed by `query-string@3.0.2` though.

**Comment by naoufal** [comment]
For anyone who came across this issue yesterday, it's worth noting that `query-string@3.0.3` resolves this.

**Comment by taion** [comment]
That's correct – this change is no longer needed for things to work. I'm merging it as being more morally correct, though.

**Comment by naoufal** [comment]
@taion Agreed. Thanks for the quick responses yesterday and your efforts with this PR. 👍

**Pull request merged**